azurelinuxagent/ga/exthandlers.py

```diff
@@ -205,6 +206,15 @@ def run_status(self):
         self.report_ext_handlers_status()
         return
 
+    def get_guid(self, name):
+        return self.last_guids.get(name, None)
+
+    def is_new_guid(self, ext_handler):
+        last_guid = self.get_guid(ext_handler.name)
+        if last_guid is None:
+            return True
+        return last_guid != ext_handler.properties.upgradeGuid
```

Comment:
How should this behave if the process crashes or vm restarts?

Reply:
@uiri Can we ever have a case where the upgradeGuid disappears? That is, it was present on the previous manifest and now is no longer? If so, then `last_guid != ext_handler.properties.upgradeGuid` will compare a GUID against `None`, treating `None` as the GUID. Is that the desired effect?

Reply:
I believe there is one potential edge case (which may not even occur in production): where a customer opts out of our new feature after observing its behavior.

There is a `None` check on `ext_handler.properties.upgradeGuid` before the call to `is_new_guid` which ensures that we get the current (pre-change) behaviour in the cases where the upgradeGuid is missing. Even without the `None` check, returning `True` would produce the desired effect.